anx/provider/provider.go

```diff
@@ -165,21 +177,26 @@ func (a anxProvider) Config() *configuration.ProviderConfig {
 	return a.config
 }
 
-func (a *anxProvider) setupProviderMetrics() {
-	a.providerMetrics = metrics.NewProviderMetrics("anexia", Version)
-	legacyregistry.MustRegister(&a.providerMetrics)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationTotalDuration)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationCreateErrorsTotal)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationDeleteRetriesTotal)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationDeleteErrorsTotal)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationCreatedTotal)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationDeletedTotal)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationCreateResources)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationPendingResources)
-	legacyregistry.MustRegister(a.providerMetrics.ReconciliationRetrievedResourcesTotal)
-
-	a.providerMetrics.MarkFeatureDisabled(featureNameLoadBalancer)
-	a.providerMetrics.MarkFeatureDisabled(featureNameInstancesV2)
+func setupProviderMetrics() metrics.ProviderMetrics {
+	providerMetrics := metrics.NewProviderMetrics("anexia", Version)
+	registerOnce.Do(func() {
```

Comment:
I'd move the `registerOnce` variable closer to the `setupProviderMetrics` function, but beside that, good catch! :+1: